tests/unittest/_torch/models/checkpoints/hf/test_weight_loader.py

```
@@ -11,6 +18,247 @@ class MyError(Exception):
+    )
+
+    assert [set(bucket) for bucket in ordinary] == [{"layers.0.weight"}]
+    assert [set(bucket) for bucket in consolidated] == [{"layers.1.weight"}]
+
+
+@pytest.mark.parametrize("use_consolidated", [False, True])
+def test_layerwise_safetensors_does_not_fallback_to_wrong_file_kind(tmp_path, use_consolidated):
+    filename = "model.safetensors" if use_consolidated else "model-consolidated.safetensors"
+    save_file({"layers.0.weight": torch.tensor([1.0])}, str(tmp_path / filename))
+
+    with pytest.raises(RuntimeError, match="requires .*safetensors weights"):
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)